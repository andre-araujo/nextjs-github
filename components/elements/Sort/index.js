import {
    Wrapper,
    Label,
    Option,
} from './styles';

const Sort = () => (
    <Wrapper>
        <Label>Sort by:</Label>

        <ul>
            <Option active>Stars</Option>
            <Option>Name</Option>
        </ul>

    </Wrapper>
);

export default Sort;
