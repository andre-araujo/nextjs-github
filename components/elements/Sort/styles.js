import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    font-size: .8rem;
    display: flex;
    user-select: none;
`;

export const Label = styled.span`
    font-family: AvantGardeDemi;
    margin-right: .6rem;
`;

export const Option = styled.li`
    font-family: AvantGardeMedium;
    display: inline-block;

    ${({ active }) => active && css`
        color: #F1912D;
    `}

    :hover {
        ${({ active }) => !active && css`
            color: #ffcc98;
            cursor: pointer;
        `}
    }

    :not(:last-child) {
        padding-right: .6rem;
        margin-right: .6rem;

        border-right: 1px solid #333;
    }
`;
