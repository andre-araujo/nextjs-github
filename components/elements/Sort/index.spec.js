import React from 'react';
import { render } from 'enzyme';

import Sort from './index';

describe('Sort component', () => {
    it('should match snapshot', () => {
        const routerMock = {
            query: {
                sort: '',
                username: '',
            },
        };

        expect(render(
            <Sort router={routerMock} />,
        )).toMatchSnapshot();
    });
});
