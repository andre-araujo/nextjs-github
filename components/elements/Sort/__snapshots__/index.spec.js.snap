// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort component should match snapshot 1`] = `
<div
  class="sc-bdVaJa eGXFjB"
>
  <span
    class="sc-bwzfXH dmLdKt"
  >
    Sort by:
  </span>
  <ul>
    <li
      class="sc-bxivhb dqXZng"
    >
      <span
        class="sc-htpNat ijxGs"
      >
        stars
      </span>
    </li>
    <li
      class="sc-bxivhb dqXZng"
    >
      <span
        class="sc-htpNat ijxGs"
      >
        forks
      </span>
    </li>
    <li
      class="sc-bxivhb dqXZng"
    >
      <span
        class="sc-htpNat ijxGs"
      >
        updated
      </span>
    </li>
  </ul>
</div>
`;
